jetty-util/src/main/java/org/eclipse/jetty/util/thread/MonitoredQueuedThreadPool.java

```diff
@@ -18,28 +18,23 @@
 
 package org.eclipse.jetty.util.thread;
 
-import java.util.concurrent.TimeUnit;
-import java.util.concurrent.atomic.AtomicInteger;
-import java.util.concurrent.atomic.AtomicLong;
-import java.util.concurrent.atomic.LongAdder;
-
-import org.eclipse.jetty.util.Atomics;
 import org.eclipse.jetty.util.BlockingArrayQueue;
 import org.eclipse.jetty.util.annotation.ManagedAttribute;
 import org.eclipse.jetty.util.annotation.ManagedObject;
 import org.eclipse.jetty.util.annotation.ManagedOperation;
+import org.eclipse.jetty.util.statistic.CounterStatistic;
+import org.eclipse.jetty.util.statistic.SampleStatistic;
 
+/**
+ * <p>A {@link QueuedThreadPool} subclass that monitors its own activity by recording queue and task statistics.</p>
+ */
 @ManagedObject
 public class MonitoredQueuedThreadPool extends QueuedThreadPool
 {
-    private final LongAdder tasks = new LongAdder();
-    private final AtomicLong maxTaskLatency = new AtomicLong();
-    private final LongAdder totalTaskLatency = new LongAdder();
-    private final MonitoringBlockingArrayQueue queue;
-    private final AtomicLong maxQueueLatency = new AtomicLong();
-    private final LongAdder totalQueueLatency = new LongAdder();
-    private final AtomicInteger threads = new AtomicInteger();
-    private final AtomicInteger maxThreads = new AtomicInteger();
+    private final CounterStatistic queueStats = new CounterStatistic();
+    private final SampleStatistic queueLatencyStats = new SampleStatistic();
+    private final SampleStatistic taskLatencyStats = new SampleStatistic();
+    private final CounterStatistic threadStats = new CounterStatistic();
```

Comment:
add these as beans and they will be added as JMX beans automagically.  No need to echo all their methods

Reply:
they will also get dumped

Reply:
Will add them as beans, but neither `CounterStatistic` nor `SampleStatistic` are annotated with JMX annotations, so they will just show up as POJOs, which will only print their toString() during `dump()`. Their methods are not exposed anyway to JMX. 
The typical usage for the additional methods in `MonitoredThreadPool` such as `getMaxBusyThreads()` is actually from embedded usage during load testing, and I don't want to expose the `*Statistic` object.

Reply:
Also, the `*Statistic` beans don't have a name to distinguish them so in the dump won't be possible to tell which bean represent what statistics without looking at the implementation.
Anyway, it's a cheap change and if we ever give the `*Statistic` beans a name they will already be dumped correctly.